driver-mapping/src/test/java/com/datastax/driver/mapping/MapperInvalidAnnotationsTest.java

```
@@ -261,27 +265,4 @@ public void setNotReadable(int i) {
 
     }
 
-    @Test(groups = "unit", expectedExceptions = IllegalArgumentException.class,
```

Why are these tests no longer applicable?  Shouldn't they still be invalid?  Also noticed that the `Invalid14` class still exists (but is no longer used)

---

It looks like in the case of unreadable it throws an NPE (on a Preconditions check) and in the case of unwritable it logs:

> [main] WARN  com.datastax.driver.mapping.config.DefaultPropertyAccessStrategy  - Property 'notWritable' is not writable and will not be mapped

but doesn't throw an exception and returns an `EntityMapper` without any columns.

---

You are right, these exceptions have became log warnings, see [here](https://github.com/datastax/java-driver/blob/4eaec3f16bb07acf3bec06d9bd51952dc436e35f/driver-mapping/src/main/java/com/datastax/driver/mapping/DefaultPropertyMapper.java#L237-L244). Not sure though if we should throw an exception or just log a warning after all.

---

I think throwing an Exception is probably a better solution as the user would likely want to be made aware that they are improperly mapping their entity.

---

Makes sense I guess. I'll fix that.